build.py

```diff
@@ -1,40 +1,521 @@
 #!/usr/bin/python3
+
+# To run:
+#    1) login as cluster admin
+#        oc login -u [cluster-admin account] -
 import subprocess
+import sys
 import re
 import time
+import json
+import pprint
+
+
+def usage_msg():
+    print(
+        """
+        Purpose:
+        
+        To build and deploy an administrative microservice to automate some administrative activities
+
+        Usage: 
+
+        build.py [project] [service] [openshift's app url] [openshift's api url] [openshift_version] [docker_file] [docker_image] [opt username] [opt password]
+
+        Examples:
+
+        build.py "acct-mgt-2" "acct-mgt" s-apps.osh.massopen.cloud "s-openshift.osh.massopen.cloud:8443" "3.11" "Dockerfile.x86" "docker.io/robertbartlettbaron/acct-mgt.x86:latest"
+        build.py "acct-mgt" "acct-mgt" k-apps.osh.massopen.cloud "k-openshift.osh.massopen.cloud:8443" "3.11" "Dockerfile.x86" "docker.io/robertbartlettbaron/acct-mgt.x86:latest" <username> <password>
+        build.py "acct-mgt" "acct-mgt" "apps.cnv.massopen.cloud" "api.cnv.massopen.cloud:6443" "4.5" "Dockerfile.x86" "docker.io/robertbartlettbaron/acct-mgt.x86:latest" <username> <password>
+    """
+    )
+
+
+def get_pod_status(project, pod_name):
+    result = subprocess.run(
+        ["oc", "-n", project, "-o", "json", "get", "pod", pod_name],
+        stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT,
+    )
+    if result.returncode == 0:
+        result_json = json.loads(result.stdout.decode("utf-8"))
+        print(result_json["status"]["phase"])
+        return result_json["status"]["phase"]
+    print("None")
+    return None
+
+
+# pass in the following parameter:
+#   project,pod_name: identify the pod
+#   statuses: the array of statuses to wait for
+def wait_while(project, pod_name, statuses, time_out=300):
+    time_left = time_out
+    time_interval = 5
+    time.sleep(time_interval)
+    status = get_pod_status(project, pod_name)
+    while status in statuses and time_left > 0:
+        time.sleep(time_interval)
+        time_left = time_left - time_interval
+        status = get_pod_status(project, pod_name)
+
+    if status in statuses:
+        return False
+    return True
+
+
+def build_docker_image(docker_file, image_name):
```

Comment:
We should replace this with automated docker builds. There's no good reason for us to be manually building and pushing images; that's a great way to end up with confusing bugs because the image doesn't match the current state of the code.

Reply:
Well, yes.  In my development, since I have been doing things local to my system in part to be able to choose which docker repository I push to, I have found this to be more convenient just for the reason you state.  If the docker image is built using the local code base, it prevents some confusing bugs.  It also gives us the flexibility to push to the docker repository that is local to OpenShift as opposed to always pushing to dockerhub or to have the automated build process on GitHub.

Reply:
Well, I did it this way as I was dealing with confusing bugs because the image did match the current state of the code.  The docker build and push are all integrated in the script so that it builds what the current state of the code on the development system is.

I'm assuming that you mean the docker builds off of GitHub as opposed to docker builds happening local to your system.